docs/ci/ci-automated-release-workflow.md

```diff
@@ -0,0 +1,120 @@
+# Creating Automated Releases: Design Document
+
+## Motivation & Overview
+
+Presently, the release processes for every library within this `instructlab` GitHub organization is entirely manual:
+
+A maintainer has to:
+
+1. Manually create a new release branch -- e.g., `release-0.y.0`,
+2. Manually create a pull request against `release-0.y.0` to cap the versions on some of the dependencies defined in their library's `requirements.txt` file,
+3. Optionally trigger an E2E test against that pull request,
+4. Wait for all pull request CI checks to complete,
+5. Manually request two maintainers to approve the pull request, and
+6. Manually create a release from the GitHub UI using that new branch
+
+This entire process takes at least 10 minutes of manual work, plus however long it takes for the pull request's checks to complete. (In some repositories, like the core repo, this can take 2+ hours.)
+
+Going forward, we should automate these release processes so that contributors and maintainers can focus more on development work and less on creating actual releases.
+
+## Automation Workflow
+
+### Brief Overview of the Automation
+
+The automation logic described in this dev-doc will be published in the form of an in-house GitHub action called `create-automated-release` and it will therefore be callable from any workflow file. Thus, it is strongly recommended that each repository maintainer creates a `.github/workflows/automated-release.yml` workflow file to call this `create-automated-release` GitHub action from.
+
+### Goals of the Automation
+
+The automated process should be:
+
+1. Configurable so that any library maintainer can configure the automation to meet their specific repository's needs
+2. Schedulable so that releases are generated according to a specific cadence
```

Comment:
No objection to this being an option but it's unclear (to me at least) whether we want to use this.

Reply:
Sometimes maintainers want to create a new y-stream release automatically every X days. For example, the core repo, `instructlab/instructlab`, has a release cadence of 14 days. 

Realistically though, the scheduling portion of this design utilizes the built-in GitHub workflow trigger conditions and is therefore independent of the design itself. Thus, if a maintainer wants to release on a cadence, they can certainly define a schedule in their workflow file, but if they  don't want to do a schedule, then they can just omit the `schedule` trigger condition.